### ISSUE TITLE
`Series.__add__` will not accept `DateOffset`

### ISSUE THREAD

**Issue by Daverball** [pandas-dev/pandas-stubs#673]
[series.pyi](https://github.com/pandas-dev/pandas-stubs/blob/main/pandas-stubs/core/series.pyi#L1379-L1388) contains overloads for most datetime like series and scalar values, but not for `DateOffset`, even though pandas appears to  support it. In most cases `Timedelta` for offsetting a`Timestamp` series is sufficient, but not when operating on calendar months/years.

**To Reproduce**
```python
import pandas as pd

s = pd.date_range('2020-01-01', '2020-02-01')
s + pd.DateOffset(months=+3)
```
mypy will report the error (`No overload variant of "__add__" of Series matches argument type "DateOffset"`)

 - OS: Linux (OpenSUSE Tumbleweed)
 - Python 3.10
 - mypy 1.0.1
 - pandas-stubs 2.0.1.230501

**Comment by Dr-Irv** [comment]
Need to add `pd.offset.BaseOffset` to addition and subtraction operators for `TimestampSeries` and `DatetimeIndex`.

**Comment by bersbersbers** [comment]
Is this example related or should I file it separately?

```python
from __future__ import annotations

import pandas as pd

def get_series() -> pd.Series[pd.Timedelta]:
    return pd.Series(pd.Timedelta(6))

series = get_series()
series /= pd.Timedelta(3)

print(series)
```

It is working fine, but I am getting "Unsupported operand types for / ("Series[Timedelta]" and "Timedelta")  [operator]" with pandas-stubs 2.0.3.230814

**Comment by hristo-mavrodiev** [comment]
> Need to add `pd.offset.BaseOffset` to addition and subtraction operators for `TimestampSeries` and `DatetimeIndex`.

This is something that needs to be added in the `pandas-stub` or something that needs to be added in the example: 
`s + pd.DateOffset(months=+3)`

Is there any workaround to trick mypy not returning an error for this example : `s + pd.DateOffset(months=+3)` ?

**Comment by Dr-Irv** [comment]
> This is something that needs to be added in the `pandas-stub` or something that needs to be added in the example:

Need a PR on pandas-stubs (with an appropriate test)

**Comment by loicdiridollou** [comment]
@Dr-Irv I believe this is actually fixed, I can't reproduce with the latest stubs (I should have fixed that in PR #1004) but please tell me if something is missing, from memory `DateOffset` inherits from `BaseOffset`.

**Comment by Dr-Irv** [comment]
> @Dr-Irv I believe this is actually fixed, I can't reproduce with the latest stubs (I should have fixed that in PR #1004) but please tell me if something is missing, from memory `DateOffset` inherits from `BaseOffset`.

Agreed.  Closing via #1004
